SU-SWS/ace-sulgryphon
# Build awards entity type with an importer and an image importing functionality

# READY FOR REVIEW

# Summary
- Provide an entity type for awards along with a migration entity to import them.
- Provide a way to download the cover images
- Provide "Site Settings" field for the API credentials.

# Steps to Test
1. checkout this branch
2. `drush @supress.local deploy`
3. configure the API credentials at `/admin/config/system/basic-site-settings`
4. run the importer `drush @supress.local mim sup_awards`
5. observe the awards list at `/admin/content/sup-award`

You can use the migration & images in your book importer by adding the following:
```
process:
  sup_book_awards:
    -
      plugin: explode
      source: work_id_number
      delimiter: /
      strict: false
    -
      plugin: entity_lookup
      value_key: sup_work_id
      entity_type: sup_award
      ignore_case: true
  [sup_book_image]:
    plugin: entity_lookup
    source: work_id_number
    value_key: sup_book_work_id
    bundle_key: bundle
    bundle: image
    entity_type: media
```
